File: configurnation-annotations/src/main/java/com/izikode/izilib/configurnationannotations/ConfigurNation.java
package com.izikode.izilib.configurnationannotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.SOURCE)
@Target(ElementType.TYPE)
public @interface ConfigurNation {

    /**
     * Config file name and SharedPreferences identifier
     * @return String
     */
    String name();

    /**
     * SharedPreferences operation mode
     * @return Integer
     */
    int mode();

}

File: configurnation-annotations/src/main/java/com/izikode/izilib/configurnationannotations/ConfigurNation.kt
package com.izikode.izilib.configurnationannotations

@Retention(AnnotationRetention.SOURCE)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
annotation class ConfigurNation(

        /**
         * Config file name and SharedPreferences identifier
         */
        val name: String,

        /**
         * SharedPreferences operation mode
         */
        val mode: Int

)
